code/modules/mob/new_player/new_player.dm

```diff
@@ -298,7 +298,7 @@
 
 /mob/new_player/proc/DiseaseCarrierCheck(var/mob/living/carbon/human/H)
 	// 10% of players are joining the station with some minor disease
-	if(prob(10))
+	if(prob(10) || isvox(H) && prob(10))
```

Comment:
Add some parantheses, you can never be too sure
if(prob(10) || (isvox(H) && prob(10)))